### ISSUE TITLE
existing picklefiles of py2 crash on py3 version (vice versa)

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#198]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/198) by Vladyslav Nenakhov (Bitbucket: [Vlad_IUP](https://bitbucket.org/Vlad_IUP), ).**

----------------------------------------

renamed from Fatal error after mss update on 26.06.17

Hello,

i just updated on my windows 7 (64bit) via "conda update mss" comand my mss. Unfurtunately i cannot start mss anymore because of a "fatal error" (see screenshot in attachment).

cmd and windows restart could not solve my problem. Do you have any suggestions? I need mss for EMeRGe campaign mission phase, which starts in ca. 2 weeks. 

Best regrads
Vlad

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Hi

the image shows that you have anaconda3 in usage.

Do you have the same problem with anaconda2? Our packages are currently only build for anaconda2. 
There is not any Py3 Package yet build https://anaconda.org/conda-forge/mss/files

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

Pickle-Files that are used for storing some settings of the GUI are currently not compatible between Py2 and Py3. They should be compatible between MSS versions.
In either case, you may delete the mss.performance.cfg/mss.sideview.cfg/mss.topview.cfg files in the configuration directory (under my Windows7 in $USER\\.config\mss).
Alternatively, you may install the last stable version of 1.4.x, which shouldn't have this issue.

I'll prepare a patch, that this error in the pickle will not crash the application.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

At least pykml is currently not compatible with py3. 

We have not tested all related libraries yet for py3. After we are finished with that we will build also a py3 package.,

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Catch ValueError when opening config-pickles

When opening a Python2 pcikle file with Python3 (and maybe vice-versa),
a ValueError may be emitted due to the different string handling.
A full solution should ideally enable the sharing of pickle files
between different Python and (to some extent) MSS versions.
A crash of the application is certainly a no go.

Resolves issue # 198

**Comment by ReimarBauer** [comment]
**Original comment by Vladyslav Nenakhov (Bitbucket: [Vlad_IUP](https://bitbucket.org/Vlad_IUP), ).**

----------------------------------------

Hello,

i deleted  mss.performance.cfg/mss.sideview.cfg/mss.topview.cfg in $USER.config\mss and mss run again. Thanks a lot Joern!

@ Reimar Bauer: i installed python 2.7 for mss as independent environment. ![170626_MSS_directory_error.jpg](https://bitbucket.org/repo/MBp6Xa/images/3638183836-170626_MSS_directory_error.jpg)

I get now in cmd always a new error msg (see attachment), because i deleted the mentioned files above. Is it normal that in the path are always  2 "\" instead of 1 "\"?

**Comment by ReimarBauer** [comment]
**Original comment by Philip Reimer (Bitbucket: [Philip Reimer](https://bitbucket.org/Philip Reimer), ).**

----------------------------------------

@Vlad_IUP : I think you meant to ping @ReimarBauer , not me ;)

**Comment by ReimarBauer** [comment]
**Original comment by Vladyslav Nenakhov (Bitbucket: [Vlad_IUP](https://bitbucket.org/Vlad_IUP), ).**

----------------------------------------

Indeed. Thanks and sorry! I changed the name.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

We are happy for anyone helping with py3. 
We will do a bug fix release next week but that is related to anaconda2 version of mss.  I hope my pull requests for pykml will be merged soon, then we can build a py3 conda version of pykml again.

The message is a "Warning" from mss, and you see a raw string representation. 


The full answer ;)
https://stackoverflow.com/questions/11924706/how-to-get-rid-of-double-backslash-in-python-windows-file-path-string

**Comment by ReimarBauer** [comment]
**Original comment by Vladyslav Nenakhov (Bitbucket: [Vlad_IUP](https://bitbucket.org/Vlad_IUP), ).**

----------------------------------------

Ok, i see. I thought double "\" could lead in the future to an error. I have to go at some point deeper in Python, ... .

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

The warnings are confusing and I will add a patch (to develop that is 1.6) to remove them in your case of starting MSS "for the first time" (i.e. without existing configurations).

However, I did not expect a "user" to start MSS from CMD since we added the icon in the Startmenu in 1.4, where you will not be able to see the logging messages. For debugging, this is obviously very helpful. (There should also be a log-file with the log in the .config/mss directory for post-mortem debugging).
